providers/flagd/README.md

```diff
@@ -35,12 +38,49 @@ func main() {
 ```
 
 ### Using flagd.FromEnv()  
-By default the flagd provider will not read environment variables to set its own configuration, however, if the `flagd.FromEnv()` option is set as an argument for the `flagd.NewProvider()` method, then the following environment variables will be checked: `FLAGD_HOST`, `FLAGD_PORT`, `FLAGD_SERVER_CERT_PATH`.
+By default the flagd provider will not read environment variables to set its own configuration, however, if the `flagd.FromEnv()` option is set as an argument for the `flagd.NewProvider()` method, then the following environment variables will be checked: `FLAGD_HOST`, `FLAGD_PORT`, `FLAGD_SERVER_CERT_PATH` & `FLAGD_CACHING_DISABLED`.
```

Comment:
I think enabling caching by default is a safe option with this primitive form of caching (caching only static resolutions). One thing to consider is future caching enhancements that we might not want on by default, or wherein we'd need more than a simple "on/off" config... it's hard to change the default later.

Reply:
Are you saying you think it's safer to disable by default in anticipation of potential future enhancements?

Reply:
I was basically thinking out loud about that question. I believe enabling it by default is still probably the right choice, I just wanted to point out that it could make future enhancements a bit more difficult.

Reply:
I'm conflicted on this too. Anyone else have an opinion? @james-milligan @beeme1mr @AlexsJones @Kavindu-Dodan

Reply:
my vote is disabled by default, if we are worried about introducing debt its better to play it safe

Reply:
I think it should be enabled by default. We'll need to consider caching for any future enhancements anyways.